HIT format multiline strings (#21758)
idaholab/moose: modules/doc/content/newsletter/2022_08.md

## MOOSE Improvements

### HIT format and parser performance improvements

The parsing performance of the HIT (hierarchial input text) file parser - the component that parses
MOOSE input files and test spec files - has been substantially improved. The parsing time now scales

> linearly with the input file length, were it previously had a square dependence. This was mostly

Reviewer: ```suggestion
linearly with the input file length, where it previously had a square dependence. This was mostly
```

noticable in extremely long input files (e.g. files generated by scripts). As an additional feature the
parser now tracks the column position of each parsed token.

The `hit format` functionality was improved to better handle the formatting of long and multi line

> strings. Multiline string can occur in input files for example as data tables for `PiecewiseLinear`

Reviewer: ```suggestion
strings. Multiline string can occur in input files when data and parameter entries 
need to be represented in a particular way - for example, as data tables for `PiecewiseLinear`
```

Reviewer: Suggestion since the sentence was lacking some context for the example.

> functions. `hit format` handled such strings poorly and disruppted careful manual formatting into

Reviewer: ```suggestion
functions. `hit format` handled such strings poorly and disrupted careful manual formatting into
```

> columns reducing the input file readability. To alleviate this two major changes were made:

Reviewer: ```suggestion
columns, reducing the input file readability. To alleviate this, two major changes were made:
```

> 1. Single quoted strings `'...'` will stay completely untouched by `hit format`. This is useful to preserve ong strings, such as PETSc options in the Executioner block, which come as parameter pairs that are often manually formatted to have corresponding key value pairs aligned to the same column positions.

Reviewer: ```suggestion
1. Single quoted strings `'...'` will stay completely untouched by `hit format`. This is useful to preserve long strings, such as PETSc options in the Executioner block, which come as parameter pairs that are often manually formatted to have corresponding key value pairs aligned to the same column positions.
```

> 2. Double quoted `"..."` strings will get formatted

Reviewer: ```suggestion
2. Double quoted `"..."` strings will get formatted in the following ways:
```

>     - Single line strings get reflowed to according to a set maximum line length (useful for long text in test spec requirements)

Reviewer: ```suggestion
    - Single line strings get reflowed according to a set maximum line length (useful for long text in test spec requirements)
```

>     - Multiline strings will only be reindented through inserting or removing spaces on the lines following the first line, to match any changes in indentation of the first line.

Reviewer: ```suggestion
    - Multiline strings will only be re-indented through inserting or removing spaces on the lines following the first line, to match any changes in indentation of the first line.
```
